sensor_test.py

```diff
@@ -0,0 +1,184 @@
+    """
+    print("Get original sensor info")
+    sensor_info_before_reboot = get_sensor_info()
+
+    print("Reboot sensor")
+    reboot_response = reboot_sensor()
+    assert reboot_response == "rebooting", "Sensor did not return proper text in response to reboot request"
+
+    print("Wait for sensor to come back online")
+    sensor_info_after_reboot = wait(
+        func=get_sensor_info,
+        condition=lambda x: isinstance(x, dict),
+        tries=10,
+        timeout=1,
+    )
+
+    print("Validate that info from Step 1 is equal to info from Step 4")
+    assert sensor_info_before_reboot == sensor_info_after_reboot, "Sensor info after reboot doesn't match sensor info before reboot"
+
+
+def test_set_sensor_name(get_sensor_info, set_sensor_name):
+    """
+    1. Set sensor name to "new_name".
+    2. Get sensor_info.
+    3. Validate that current sensor name matches the name set in Step 1.
+    """
+    # Step 1: Set sensor name to "new_name"
+    set_sensor_name("new_name")
+    
+    # Step 2: Get sensor_info
+    sensor_info = get_sensor_info()
+    
+    # Step 3: Validate that current sensor name matches the name set in Step 1
+    assert sensor_info['name'] == "new_name", f"Sensor name should be 'new_name', but got '{sensor_info['name']}'"
+
+
+def test_set_sensor_reading_interval(
+    get_sensor_info, set_sensor_reading_interval, get_sensor_reading
+):
+    """
+    1. Set sensor reading interval to 1.
+    2. Get sensor info.
+    3. Validate that sensor reading interval is set to interval from Step 1.
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
+
+    """
+    Test to set sensor reading interval to 1, get sensor info, validate that the sensor reading interval is set,
+    get sensor reading, wait for the interval, get sensor reading again, and validate that the readings are different.
+    """
+    # Step 1: Set sensor reading interval to 1
+    set_sensor_reading_interval(1)
+    
+    # Step 2: Get sensor info
+    sensor_info = get_sensor_info()
+    
+    # Step 3: Validate that sensor reading interval is set to interval from Step 1
+    assert sensor_info['reading_interval'] == 1, f"Sensor reading interval should be 1, but got {sensor_info['reading_interval']}"
+    
+    # Step 4: Get sensor reading
+    initial_reading = get_sensor_reading()
+    
+    # Step 5: Wait for interval specified in Step 1
+    # (Note: This step is simulated with a sleep of 1 second)
+    time.sleep(1)
+    
+    # Step 6: Get sensor reading
+    new_reading = get_sensor_reading()
+    
+    # Step 7: Validate that reading from Step 4 doesn't equal reading from Step 6
+    assert initial_reading != new_reading, "Sensor readings should be different after the interval"
+
+
+
+# Максимальна версія прошивки сенсора -- 15
+def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
```

Comment:
`test_update_sensor_firmware()` is defined here and on line 138 but there should be only one definition.